ai/src/ai/common/executor.py

```diff
@@ -198,15 +218,39 @@ def get_provider_executor(self) -> ProviderExecutor:
     )
     return provider_executor
 
-  def execute(self, input: ExampleInput):
-    return self.get_provider_executor().execute(input)
+  def execute(self, input: ExampleInput, seed: int | None = None):
+    provider_executor = self.get_provider_executor()
+    if seed is None:
+      return provider_executor.execute(input, seed=seed)
+
+    # Create a unique cache key
+    cache_key = md5(
```

Comment:
Would there be scenarios where we want to clear the cache or subsets of the cache? What would be the steps to do so?

Reply:
I think there's two reasons to clear the cache: 1) you want to simply re-generate new samples (e.g. temp>0) in which case you could use the seed parameter (as part of the eval creation) or 2) the cache contains buggy outputs in which case you could either modify the cache key (e.g. at first I wasn't capturing all the inputs used into the cache key) or you could delete the entire folder or subsets of the `.cache` dir in the `data` dir